main.go

```diff
@@ -475,22 +475,93 @@ func makeSecretChanges(kubeClient *k8s.Client, secret *corev1.Secret, initiator
 	return status, nil
 }
 
-func processSecret(kubeClient *k8s.Client, secret *corev1.Secret, initiator string) (status string, err error) {
+func isEventExist(kubeClient *k8s.Client, namespace string, name string, event *corev1.Event ) (exist bool,err error){
+	err = kubeClient.Get(context.Background(), namespace, name, event)
+	exist = true
+	if apiErr, ok := err.(*k8s.APIError); ok {
+		if apiErr.Code == http.StatusNotFound {
+			exist = false
+			return exist,nil
+		}
+	}
+	return exist,nil
+}
 
-	status = "failed"
+func postEventAboutStatus(kubeClient *k8s.Client, secret *corev1.Secret, eventType string, action string, reason string, kind string, reportingController string, reportingInstance string)(err error){
+	log.Info().Msgf(" Starting Function")
+	now := time.Now()
+	secs := int64(now.Unix())
+	count := int32(1)
+	var eventResp corev1.Event
+	var exist bool
+	exist, err = isEventExist(kubeClient, *secret.Metadata.Namespace, *secret.Metadata.Name, &eventResp)
+	if exist {
+		count = *eventResp.Count+ 1
+		eventResp.Type = &eventType
+		eventResp.Action = &action
+		eventResp.Reason = &reason
+		eventResp.Count = &count
+		eventResp.LastTimestamp.Seconds = &secs
+		err = kubeClient.Update(context.Background(), &eventResp)
+		if err != nil {
+			log.Error().Msgf("Event %v.%v - Updating Event has an error.\n\t%s", *eventResp.Metadata.Name, *eventResp.Metadata.Namespace, err.Error())
+			return err
+		}
+		log.Info().Msgf("Event %v.%v - has been updated successfully...", *eventResp.Metadata.Name, *eventResp.Metadata.Namespace)
+		return
+	}
+	event := new(corev1.Event)
```

Comment:
White line after closing bracket makes it more readable.

Reply:
sure thing!